udf/udf_enforce_date_param.sql

```diff
@@ -0,0 +1,39 @@
+/*
+
+Returns true if the two passed dates are identical or if either of them is
+equal to the special flag date 0001-01-01.
+
+This is useful for queries that have a @submission_date parameter and are
+generally executed incrementally, but occasionally need to be run over all
+time to recreate the table. A query can be written with a WHERE clause like:
+
+    WHERE udf_enforce_date_param(submission_date, @submission_date)
+
+Under normal operation, this clause is equivalent to:
+
+    WHERE submission_date = @submission_date
+
+But if you execute the query with `--parameter submission_date:DATE:1111-11-11`,
+this essentially removes the WHERE clause.
```

Comment:
could you add justification for using a magic date value here, instead of, for instance, `NULL`

Reply:
I would much prefer to use NULL, but I was assuming it's not possible to pass null from the command line. If I'm wrong, that would be great.

Potentially a simpler alternative is to rely on a wrapper script instead that looks for `submission_date = @submission_date` and replaces with `TRUE` and then we don't have a need for a magic value, or have to worry about rewriting queries to support this UDF.

Reply:
> I was assuming it's not possible to pass null from the command line

you can do `--parameter=submission_date:DATE:NULL`

Also, it turns out if the `--parameter` flag is missing for a given parameter the cli throws `Error in query string: Error processing job '...': Undeclared query parameters`, so it is safe to use `NULL`

I suggest we do this `(submission_date = @submission_date OR @submission_date IS NULL)` rather than using a UDF, so that we don't worry about parameter order, obfuscation, parameter order, or unintentionally including `WHERE submission_date IS NULL` on daily results

Reply:
This sounds great. Thanks for investigating. I will close this PR in favor of the `(submission_date = @submission_date OR @submission_date IS NULL)` solution.